c++/src/CMakeLists.txt

```diff
@@ -202,18 +202,29 @@ endif(BUILD_ENABLE_AVX512)
 add_library (orc STATIC ${SOURCE_FILES})
 
 target_link_libraries (orc
-  orc::protobuf
-  orc::zlib
-  orc::snappy
-  orc::lz4
-  orc::zstd
-  ${LIBHDFSPP_LIBRARIES}
+  INTERFACE
+    $<INSTALL_INTERFACE:protobuf::libprotobuf>
+    $<INSTALL_INTERFACE:ZLIB::ZLIB>
+    $<INSTALL_INTERFACE:Snappy::snappy>
+    $<INSTALL_INTERFACE:$<TARGET_NAME_IF_EXISTS:LZ4::lz4_static>>
```

Comment:
Adding `TARGET_NAME_IF_EXISTS` generator expression looks like a bad idea as it does not complain a missing call to find_package any more.